app/src/main/java/com/example/next_show/adapters/ShowAdapter.java

```diff
@@ -55,6 +56,23 @@ public void clear() {
         notifyDataSetChanged();
     }
 
+    // uses basic search algorithm
+    public int search(String id){
+        for (int index = 0; index < getItemCount(); index++){
+            Show s = shows.get(index);
+            if(s.getId().equals(id)){
+                return index;
+            }
+        }
+        return -1;
```

Comment:
nit: we could save this into a constant like NOT_FOUND

Reply:
There also might be some api on the adapter itself that you could utilize -- can be a follow up diff, but can you take a look?